src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -12500,6 +13390,363 @@ public void SynthesizedDelegateTypes_ParamsArray_DifferentElementTypes()
+                var d = new D().M;
+                System.Console.WriteLine(d.GetType());
+
+                public class D : C
+                {
+                    public override void M(int[] xs) { }
+                }
+
+                public abstract class C
+                {
+                    public abstract void M(params int[] xs);
+                }
+                """;
+            var verifier = CompileAndVerify(source,
+                expectedOutput: """
+                <>f__AnonymousDelegate0`1[System.Int32]
+                <>f__AnonymousDelegate0`1[System.Int32]
+                """).VerifyDiagnostics();
+            var cm = verifier.Compilation.GetMember<IMethodSymbol>("C.M");
+            Assert.True(cm.Parameters.Single().IsParams);
+            var dm = verifier.Compilation.GetMember<IMethodSymbol>("D.M");
+            Assert.True(dm.Parameters.Single().IsParams);
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_ParamsArray_LeastOverriddenMethod_Different()
+        {
+            var source1a = """
+                public abstract class C
+                {
+                    public abstract void M(int[] xs);
+                }
+                """;
+            var comp1a = CreateCompilation(source1a, assemblyName: "Lib1").VerifyDiagnostics();
+            var comp1aRef = comp1a.EmitToImageReference();
+
+            var source2 = """
+                public class D : C
+                {
+                    public override void M(int[] xs) { }
+                }
+                """;
+            var comp2 = CreateCompilation(source2, [comp1aRef]).VerifyDiagnostics();
+            var comp2Ref = comp2.EmitToImageReference();
+
+            var source3 = """
+                var c = ((C)new D()).M;
+                System.Console.WriteLine(c.GetType());
+                var d = new D().M;
+                System.Console.WriteLine(d.GetType());
+                """;
+            var verifier = CompileAndVerify(source3, [comp2Ref, comp1aRef],
+                expectedOutput: """
+                System.Action`1[System.Int32[]]
+                System.Action`1[System.Int32[]]
+                """).VerifyDiagnostics();
+
+            var cm = verifier.Compilation.GetMember<IMethodSymbol>("C.M");
+            Assert.False(cm.Parameters.Single().IsParams);
+            var dm = verifier.Compilation.GetMember<IMethodSymbol>("D.M");
+            Assert.False(dm.Parameters.Single().IsParams);
+
+            var source1b = """
+                public abstract class C
+                {
+                    public abstract void M(params int[] xs);
+                }
+                """;
+            var comp1b = CreateCompilation(source1b, assemblyName: "Lib1").VerifyDiagnostics();
+            verifier = CompileAndVerify(source3, [comp2Ref, comp1b.EmitToImageReference()],
+                expectedOutput: """
+                <>f__AnonymousDelegate0`1[System.Int32]
```

Comment:
><>f__AnonymousDelegate0`1[System.Int32] [](http://example.com/codeflow?start=16&length=39)

Please explicitly assert `params` and/or default values (depending what aspect we are focusing on) on synthesized delegates.